### PR TITLE
TSDB: Fix time series bounds default value error

### PR DESCRIPTION
**Pull request by weizijun**
The index.mode validation is use the setting's validator mechanism.
```
    public static final Setting<IndexMode> MODE = Setting.enumSetting(
        IndexMode.class,
        "index.mode",
        IndexMode.STANDARD,
        new Setting.Validator<IndexMode>() {
            @Override
            public void validate(IndexMode value) {}

            @Override
            public void validate(IndexMode value, Map<Setting<?>, Object> settings) {
                value.validateWithOtherSettings(settings);
            }

            @Override
            public Iterator<Setting<?>> settings() {
                return IndexMode.VALIDATE_WITH_SETTINGS.iterator();
            }
        },
        Property.IndexScope,
        Property.Final
    );
```
The `settings` method will init the settings that are not configured in the setting builder and the not configured settings are use the default value, so if set the setting to default value is the same with not put it in the builder.

```
        private void settingRequiresTimeSeries(Map<Setting<?>, Object> settings, Setting<?> setting) {
            if (Objects.equals(setting.getDefault(Settings.EMPTY), settings.get(setting))) {
                 throw new IllegalArgumentException("[" + IndexSettings.MODE.getKey() + "=time_series] requires [" + setting.getKey() + "]");
             }
         }
```

This method check the setting if configured, but if set the default value, it also meet the equals check.
To solve the problem, I move the validator from the setting validator mechanism to explicitly called in IndexSettings constructor.

And this change will modify the validation of empty routing_path, so I add it in the TIME_SERIES mode's `validateWithOtherSettings`

it fixed the bug of https://github.com/elastic/elasticsearch/issues/82832

### CODE REVIEW

**Comment by csoulios** [comment]
@elasticmachine test this please

File: server/src/main/java/org/elasticsearch/index/IndexMode.java [hunk 1]
```diff
@@ -78,22 +73,26 @@ public MetadataFieldMapper buildTimeSeriesIdFieldMapper() {
...
-                if (false == Objects.equals(unsupported.getDefault(Settings.EMPTY), settings.get(unsupported))) {
+                if (settings.hasValue(unsupported.getKey())) {
                     throw new IllegalArgumentException(error(unsupported));
                 }
             }
             settingRequiresTimeSeries(settings, IndexMetadata.INDEX_ROUTING_PATH);
             settingRequiresTimeSeries(settings, IndexSettings.TIME_SERIES_START_TIME);
             settingRequiresTimeSeries(settings, IndexSettings.TIME_SERIES_END_TIME);
+
+            if (IndexMetadata.INDEX_ROUTING_PATH.get(settings).isEmpty()) {
```
Comment (imotov) [inline server/src/main/java/org/elasticsearch/index/IndexMode.java:89]: Nice catch!

Comment (weizijun) [reply server/src/main/java/org/elasticsearch/index/IndexMode.java:89]: Thanks!

**Comment by weizijun** [comment]
> Could you explain rationale for this change? Seems like you are circumventing the standard way of checking the compatibly with some special way that we need to remember to call. I am not quite sure I can come up with a good reason for that.

@imotov As I say in the PR content, The Setting.Validator will init the not configured settings in the settings() method, so the settings in the IndexMode.VALIDATE_WITH_SETTINGS will be inited, even if users not set the these settings.
index.time_series.start_time and index.time_series.end_time will set the default in the Validator settings() method, even if users not set. So check the default equals will failed, and to check settings's hasValue method will also true , as the settings is inited in the the Validator settings() method.

So I move the validateWithOtherSettings call from the Setting.Validator to the constructor of IndexSettings

File: server/src/main/java/org/elasticsearch/index/IndexSettings.java [hunk 2]
```diff
@@ -518,20 +518,7 @@ public void validate(Instant value, Map<Setting<?>, Object> settings) {
...
-            public void validate(IndexMode value, Map<Setting<?>, Object> settings) {
-                value.validateWithOtherSettings(settings);
-            }
-
-            @Override
-            public Iterator<Setting<?>> settings() {
-                return IndexMode.VALIDATE_WITH_SETTINGS.iterator();
-            }
-        },
+        v -> {},
```
[the remark this answers is not in the document]
Comment (weizijun) [reply server/src/main/java/org/elasticsearch/index/IndexSettings.java:521]: As I say in the PR content, The Setting.Validator will init the not configured settings in the settings() method, so the settings in the `IndexMode.VALIDATE_WITH_SETTINGS` will be inited, even if users not set the these settings.
`index.time_series.start_time` and `index.time_series.end_time` will set the default in the Validator settings() method, even if users not set. So check the default equals will failed, and to check settings's `hasValue` method will also true , as the settings is inited in the the Validator settings() method.

So I move the `validateWithOtherSettings` call from the Setting.Validator to the constructor of `IndexSettings`

**Comment by weizijun** [comment]
https://github.com/elastic/elasticsearch/pull/83236 replaces this PR.

**Pull request closed without merging**